commercial/app/commercial/CommercialLifecycle.scala

```diff
@@ -22,12 +23,43 @@ trait CommercialLifecycle extends GlobalSettings with Logging with ExecutionCont
     MoneyBestBuysRefresh
   )
 
-  private def recordEvent(feedName: String, eventName: String, maybeDuration: Option[Duration]): Unit = {
-    val key = s"${feedName.toLowerCase.replaceAll("[\\s/]+", "-")}-$eventName-time"
-    val duration = maybeDuration map (_.toMillis.toDouble) getOrElse -1d
-    CommercialMetrics.metrics.put(Map(s"$key" -> duration))
+
+  private lazy val metricMap = Map(
+    "fetch-failure" -> AkkaAgent(0.0),
+    "fetch-success" -> AkkaAgent(0.0),
+    "parse-failure" -> AkkaAgent(0.0),
+    "parse-success" -> AkkaAgent(0.0)
+  )
+
+
+  private def recordSuccess(eventName:String): Unit = {
+    val keyName = s"$eventName-success"
+    metricMap
+      .get(keyName)
+      .foreach(agent => agent.send(_ + 1.0))
   }
 
+  private def recordFailure(eventName:String): Unit = {
+    val keyName = s"$eventName-failure"
+    metricMap
+      .get(keyName)
+      .foreach(agent => agent.send(_ + 1.0))
+  }
+
+
+  private def updateFetchAndParseMetrics(): Unit = {
```

Comment:
I would give this method a more generic name like `updateMetrics` because if we decide later to add some more metrics you will add them to the Map above and then this method name would not reflect what it is actually doing.